### DISCUSSION TITLE
Ability to add comment to a zip file

### DISCUSSION THREAD

**Discussion by ra-dave** [rikyoz/bit7z#140] [Ideas]
Zip files have the ability to have a comment string.  7-Zip supports this through its GUI but I don't see any option to do this from the command line and I don't know if the API allows it.

It would be nice to have this functionality in bit7z, if it's possible.  For reference this is the API doc for this from minizip-ng:

mz_zip_set_comment
Sets the zip file's global comment string when the zip file is opened for writing. According to the zip file specification, each zip file can store a global comment with a maximum length of UINT16_MAX or 65535 characters.

**Comment by rikyoz** [comment]
I did a bit of research, and I could not find any CLI option too.

I also searched through the 7-zip source code and might have found how it handles archive comments, though I still need to figure out how to exploit it in the library.
Bit7z supports reading the comment (as it is like any other property) but not writing it.

It's a helpful feature, so I'll definitely work on supporting it!
Thanks for pointing it out and for using bit7z! 🙏
